config.go

```diff
@@ -131,6 +136,53 @@ func (log Logger) LoadConfiguration(filename string) {
 	}
 }
 
+/* 
+   Replace all instances of `${var}` in the string with the value of the environment variable `var`.
+   The literals `$` and `\` may be escaped with a backspace. Examples:
+     `${var}` becomes `<value of var>`
+     `\${var}` becomes `${var}`,
+     `\\${var}` becomes `\<value of var>`
+*/
+func substituteEnv(prop string) string {
+	vStart := 0
+	state := 1
```

Comment:
nit: This would be a little bit easier to follow if you used symbolic names for the states rather than integers. Maybe a comment would be sufficient?